### PR TITLE
Remove analyze --dartdocs flag

### LINKED ISSUE #79798: Remove the --dartdocs flag from `flutter analyze`
This flag is confusing. If someone has specified `public_member_api_docs`, we should just be showing the items missing docs. It is even more confusing in that the output ends up saying:

```
Analyzing flutter_svg...                                                
No issues found! (ran in 2.3s; 2 public members lack documentation)
```

My eyes just see "no issues found" when I know I have public members missing docs, and this is causing me to miss that. 

@jonahwilliams can we just remove this, or would we need to deprecate it and turn it into a no-op?

### PR DESCRIPTION
**Pull request by jmagman**
Fixes https://github.com/flutter/flutter/issues/79798

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: dev/devicelab/bin/tasks/dartdocs.dart [hunk 1]
```diff
@@ -2,70 +2,13 @@
...
-      throw Exception('flutter analyze had unexpected output (we saw $otherLines unexpected line${ otherLines == 1 ? "" : "s" })');
-    final Map<String, dynamic> data = <String, dynamic>{
-      'members_missing_dartdocs': publicMembers,
-      'analysis_errors': otherErrors,
-      'elapsed_time_ms': clock.elapsedMilliseconds,
-    };
-    return TaskResult.success(data, benchmarkScoreKeys: data.keys.toList());
+    // TODO(jmagman): Remove once dartdocs builder can be deleted
+    // when https://github.com/flutter/flutter/issues/79798 rolls to stable.
+    return TaskResult.success(null);
```
Comment (dnfield) [inline dev/devicelab/bin/tasks/dartdocs.dart:12]: I was going to say that I thought we'd still need this, but I guess we're benchmarking dartdocs for the repo under the regular analysis task?

**Comment by dnfield** [comment]
We need https://github.com/flutter/engine/pull/25615 first. I check plugins, plugin_tools, and recipes - no mention of --dartdocs in those.

Rather than completely removing the flag, do we want to perhaps print a deprecation warning indicating that it's now a no-op and will be removed in the next stable release?

**Comment by jmagman** [comment]
> Rather than completely removing the flag, do we want to perhaps print a deprecation warning indicating that it's now a no-op and will be removed in the next stable release?

Good idea.

**Comment by jmagman** [comment]
> > Rather than completely removing the flag, do we want to perhaps print a deprecation warning indicating that it's now a no-op and will be removed in the next stable release?
> 
> Good idea.

Actually, what we've done in the past is to make it a no-op and add a deprecation note to the help text.  You're suggesting we keep it working but print a warning?

**Comment by dnfield** [comment]
No, just print the warning and make it do nothing else

**Comment by jmagman** [comment]
> No, just print the warning and make it do nothing else

I re-added the flag and added the deprecation note to the help text.  Do you also think the printed warning is necessary?

**Review by dnfield — APPROVED** [review]
Lgtm

**Pull request merged**